xc/xenops_server_xen.ml

```diff
@@ -2037,13 +2028,7 @@ module VM = struct
       debug "vm %s: persisting metadata %s" k state_new;
       (if state_new <> state then debug "vm %s: different original metadata %s" k state)
     ;
-    let _ = DB.update vm.Vm.id (fun d ->
-        let non_persistent = match d with
-          | None -> with_xc_and_xs (fun xc xs -> generate_non_persistent_state xc xs vm persistent)
-          | Some vmextra -> vmextra.VmExtra.non_persistent
-        in
-        Some { VmExtra.persistent = persistent; VmExtra.non_persistent = non_persistent; }
-      )
+    let _ = DB.update vm.Vm.id (fun d -> Some { VmExtra.persistent = persistent })
```

Comment:
I think you can do `Some VmExtra.{persistent}` here

Reply:
Fixed.